beetsplug/edit.py

```diff
@@ -151,6 +154,23 @@ def __init__(self):
             'ignore_fields': 'id path',
         })
 
+        self.register_listener('before_choose_candidate',
+                               self.before_choose_candidate_listener)
+        self.register_listener('import_begin', self.import_begin_listener)
+
+    def _set_reference_field(self, field):
+        """Set the "unequivocal, non-editable field" that will be used for
+        reconciling back the user changes.
+        """
+        if field == 'id':
+            self.reference_field = 'id'
+            self.ref_field_value = lambda o: int(o.id)
+            self.obj_from_ref = lambda d: int(d['id'])
+        elif field == 'path':
+            self.reference_field = 'path'
+            self.ref_field_value = lambda o: util.displayable_path(o.path)
+            self.obj_from_ref = lambda d: util.displayable_path(d['path'])
```

Comment:
Hmm... It's unfortunate that this needs to update fields on the plugin object. Is there any way we can control this instead by just passing a parameter to the relevant functions? That should make the logical flow easier to read (and, hopefully, it should make for easier testing).